2015/Call.js

```diff
@@ -1,19 +1,13 @@
 'use strict';
 
 var GetIntrinsic = require('../GetIntrinsic');
+var callBound = require('../helpers/callBound');
 
-var $TypeError = GetIntrinsic('%TypeError%');
-
-var inspect = require('object-inspect');
-
-var IsCallable = require('./IsCallable');
+var $apply = GetIntrinsic('%Reflect.apply%', true) || callBound('%Function.prototype.apply%');
```

Comment:
It's not possible to *both* call-bind call and apply in a pre-ES5 environment. There's a reason I stuck with `F.apply`.

In other words, `apply` can't ever be fully robust.

Reply:
In the **ES3** case, the resulting function is equivalent to:

```js
GetIntrinsic('%Function.prototype.apply%').apply(F, V, args);
```

So it only depends on `%Function.prototype.apply%`’s `apply` property resolving to `%Function.prototype.apply%`, which is a lot more reliable than some random `F` object not having an overridden `apply` property (e.g.: `callBind`).